tools/k8s-test/wait.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"time"
+)
+
+func WaitForObjects(ctx context.Context, objects []Object) error {
+	for _, object := range objects {
+		switch object.Kind {
+		case "Deployment":
+			if err := WaitForDeployment(ctx, object.Name); err != nil {
+				return err
+			}
+		case "DaemonSet":
+			if err := WaitForDaemonSet(ctx, object.Name); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func WaitForDeployment(ctx context.Context, name string) error {
+	replicaSets, err := GetReplicaSetsByOwner(ctx, Object{Kind: "Deployment", Name: name})
+	if err != nil {
+		return err
+	}
+
+	for _, replicaSet := range replicaSets {
+		if err := WaitForReplicaSet(ctx, replicaSet.Name); err != nil {
+			return err
+		}
+	}
+
+	return waitFor(ctx, func(ctx context.Context) (bool, error) {
+		deployment, err := GetDeployment(ctx, name)
+		if err != nil {
+			return false, err
+		}
+		if err := CheckDeploymentReady(deployment); err != nil {
+			Warnln("deployment %q is not ready yet: %v", name, err.Error())
+			return false, nil
+		}
+		Goodln("deployment %q is ready", name)
+		return true, nil
+	})
+}
+
+func WaitForReplicaSet(ctx context.Context, name string) error {
+	pods, err := GetPodsByOwner(ctx, Object{Kind: "ReplicaSet", Name: name})
+	if err != nil {
+		return err
+	}
+
+	for _, pod := range pods {
+		if err := WaitForPod(ctx, pod.Name); err != nil {
+			return err
+		}
+	}
+
+	return waitFor(ctx, func(ctx context.Context) (bool, error) {
+		replicaSet, err := GetReplicaSet(ctx, name)
+		if err != nil {
+			return false, err
+		}
+		if err := CheckReplicaSetReady(replicaSet); err != nil {
+			Warnln("replica set %q is not ready yet: %v", name, err.Error())
+			return false, nil
+		}
+		Goodln("replica set %q is ready", name)
+		return true, nil
+	})
+}
+
+func WaitForDaemonSet(ctx context.Context, name string) error {
+	pods, err := GetPodsByOwner(ctx, Object{Kind: "DaemonSet", Name: name})
```

Comment:
"Kinds" are used in different places, maybe we can create constants to in a common place and use them, in that way it will be easier to add/update/delete them in the future

Reply:
sounds good. will do.